specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json

```diff
@@ -0,0 +1,269 @@
+        "operationId": "ServerConnectionPolicies_ListByServer",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved the server connection policy list",
+            "schema": {
+              "$ref": "#/definitions/ServerConnectionPolicyListResult"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 400 InvalidConnectionPolicyName - Invalid connection policy name.\n\n * 400 ConnectionPolicyParamsNull - The request body cannot be null.\n\n * 400 InvalidConnectionPolicyType - Connection type is invalid\n\n * 404 OperationIdNotFound - The operation with Id does not exist.\n\n * 404 OperationIdNotFound - The operation with Id does not exist.\n\n * 404 SubscriptionDoesNotHaveServer - The requested server was not found\n\n * 404 ServerNotInSubscriptionResourceGroup - Specified server does not exist in the specified resource group and subscription.\n\n * 409 OperationCancelled - The operation has been cancelled by user.\n\n * 409 OperationInterrupted - The operation on the resource could not be completed because it was interrupted by another operation on the same resource.\n\n * 500 OperationTimedOut - The operation timed out and automatically rolled back. Please retry the operation."
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Lists a servers connection policies": {
+            "$ref": "./examples/ServerConnectionPoliciesList.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ServerConnectionPolicyProperties": {
+      "description": "The properties of a server connection policy.",
+      "required": [
+        "connectionType"
+      ],
+      "type": "object",
+      "properties": {
+        "connectionType": {
+          "description": "The server connection type.",
+          "enum": [
+            "Default",
+            "Redirect",
+            "Proxy"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "ServerConnectionType",
+            "modelAsString": false
+          }
+        }
+      }
+    },
+    "ServerConnectionPolicy": {
+      "description": "A server connection policy",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "location": {
```

Comment:
Usually nested proxy resources shouldn't need a location property.

Reply:
it is intended to prevent breaking change from 2014-04-01.

Reply:
How so, I thought this is a new resource type you're adding in this version?

Reply:
Hi Tim, connection policies is a resource that we already support in the 2014 version. However we had to introduce it here as part of this new API to allow for async behavior because we had some ICM in the past which led us to deciding this API should be Async. With this change the idea is customers will have the option to consume the new ask to call this api without having to chance any of their code, and so we've tried to keep it matching with the 2014 version.

Reply:
@tunafish88 Thanks, makes sense.